src/nupic/algorithms/Connections.cpp

```diff
@@ -199,17 +225,33 @@ void Connections::destroySynapse(Synapse synapse) {
   segmentData.synapses.erase(synapseOnSegment);
 
   destroyedSynapses_.push_back(synapse);
+
+  if( synapseData.permanence > connectedThreshold_ - EPSILON )
+    segmentData.numConnected--;
 }
 
 void Connections::updateSynapsePermanence(Synapse synapse,
                                           Permanence permanence) {
   permanence = std::min(permanence, maxPermanence );
   permanence = std::max(permanence, minPermanence );
-  synapses_[synapse].permanence = permanence;
 
-  // for (auto h : eventHandlers_) {
-  //   h.second->onUpdateSynapsePermanence(synapse, permanence);
```

Comment:
a test is failing because of this, we can either: 
- remove the test
- enable it here (how much of an impact this optimization had?) 
- enable for `#ifndef NDEBUG`

Reply:
...ok again, handlers handeled below: https://github.com/htm-community/nupic.cpp/pull/153/files/a428461a7458d60112d6acbd7943eb0911ec77ba..2ccdb9992738866337b4957ba71702fb486684aa#diff-dc7539503f4135f1c566ef5ba871fc7fR252